### ISSUE TITLE
Upgrade to ldapjs v3

### ISSUE THREAD

**Issue by shaozi** [shaozi/ldap-authentication#50]
ldapjs v3 has breaking changes on the result object returned from `searchEntry` event callback. Instead of have `.object` field in the `entry` parameter, it is now a `.pojo` field and the structure of it is different.

I have created a branch to handle this change. hopefully it does not cause issue.

**Comment by shaozi** [comment]
@christopherpickering This change may affect the latest `.raw` feature you have added. Can you check out this branch with the latest ldapjs v3 to see how to add `.raw` there? (I just assign the ldapjs entry to it. probably not what you expect)

New branch is here: https://github.com/shaozi/ldap-authentication/tree/upgrade-ldapjs-3

**Comment by shaozi** [comment]
@christopherpickering That sounds great. I am going to include your case to the README so people will be able to understand and use it. Can you give a simple example for the usage of `;binary` in ldap-authentication?

Thanks!

**Comment by christopherpickering** [comment]
yep, thanks! something along these lines:

```ts
export async function verifyLogin(email: string, password: string) {

  const options = {
   ...other config options
    userPassword: password,
    username: email,
    attributes: ['thumbnailPhoto;binary', 'givenName', 'sn', 'sAMAccountName', 'userPrincipalName', 'memberOf' ]
  };

  try {
    const ldapUser = await authenticate(options);

    if (!ldapUser) {
      return { error: "user not found" };
    }

    // accessing the image
    const profilePhoto = ldapUser['thumbnailPhoto;binary'];
        
    /* using the image
	<img src={`data:image/*;base64,${user.profilePhoto}`} />
    */
    return { user: ldapUser };
  }
}


```

**Comment by shaozi** [comment]
Released in #51
